src/pcm/SoxrResampler.cxx

```diff
@@ -80,20 +95,114 @@ soxr_parse_quality(const char *quality) noexcept
 	return SOXR_INVALID_RECIPE;
 }
 
+static unsigned
+SoxrParsePrecision(unsigned value) {
+	switch (value) {
+		case 16:
+		case 20:
+		case 24:
+		case 28:
+		case 32:
+			break;
+		default:
+			throw FormatInvalidArgument(
+				"soxr converter invalid precision : %d [16|20|24|28|32]", value);
+			break;
+	}
+	return value;
+}
+
+static double
+SoxrParsePhaseRespons(unsigned value) {
+	if (value > 100) {
+		throw FormatInvalidArgument(
+			"soxr converter invalid phase_respons : %d (0-100)", value);
+	}
+
+	return double(value);
+}
+
+static double
+SoxrParsePassbandEnd(const char *svalue) {
+	char *endptr;
+	double value = strtod(svalue, &endptr);
+	if (strlen(svalue) == 0 || *endptr) {
```

Comment:
This is a clumsy check - three things are wrong here:
- it use `strlen()` to check if a string is empty, but you could better use `*svalue == 0`
- if you check `svalue`, you better do it before calling `strtod()`, because this doesn't depend on `strtod()`
- ... but even that would be complicated; the canonical way to rule out a parsing error here is checking `svalue == endptr` because that will tell you whether anything has been parsed at all